src/mmds/src/token_headers.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use micro_http::{HttpHeaderError, RequestError};
+use std::collections::HashMap;
+use std::result::Result;
+
+/// Wrapper over the list of token headers associated with a Request.
+#[derive(Debug, PartialEq)]
+pub struct TokenHeaders {
+    /// The `X-metadata-token` header might be used by HTTP clients to specify a token in order
+    /// to authenticate to the session. This is used for guest requests to MMDS only.
+    x_metadata_token: Option<String>,
+    /// The `X-metadata-token-ttl-seconds` header might be used by HTTP clients to specify
+    /// the expiry time of a token. This is used for PUT requests issued by the guest to MMDS only.
+    x_metadata_token_ttl_seconds: Option<u32>,
+}
+
+impl Default for TokenHeaders {
+    /// Token headers are not present in the request by default.
+    fn default() -> Self {
+        Self {
+            x_metadata_token: None,
+            x_metadata_token_ttl_seconds: None,
+        }
+    }
+}
+
+impl TokenHeaders {
+    /// `X-metadata-token` header.
+    const X_METADATA_TOKEN: &'static str = "X-metadata-token";
+    /// `X-metadata-token-ttl-seconds` header.
+    const X_METADATA_TOKEN_TTL_SECONDS: &'static str = "X-metadata-token-ttl-seconds";
+
+    /// Return `TokenHeaders` from headers map.
+    pub fn try_from(map: &HashMap<String, String>) -> Result<TokenHeaders, RequestError> {
+        let mut headers = Self::default();
+
+        if let Some(token) = map.get(TokenHeaders::X_METADATA_TOKEN) {
+            headers.x_metadata_token = Some(token.to_string());
+        }
```

Comment:
slightly more rustacean:

```suggestion
        map.get(TokenHeaders::X_METADATA_TOKEN).map(|token| {
            headers.x_metadata_token = Some(token.to_string());
        });
```

Reply:
Given https://github.com/firecracker-microvm/firecracker/pull/2790#discussion_r753137956, I would prefer to keep the `if` construction for the `X_METADATA_TOKEN` header as well, for consistency reasons. WDYT?